tardis/tests/tests_slow/conftest.py

```diff
@@ -5,34 +5,53 @@
 import pytest
 from astropy import units as u
 
+from tardis import __githash__ as tardis_githash
 from tardis.tests.tests_slow.report import DokuReport
 from tardis.tests.tests_slow.plot_helpers import PlotUploader
 
 
+def pytest_addoption(parser):
+    parser.addoption("--integration-tests", dest="integration-tests", default=None,
+                     help="path to configuration file for integration tests")
+    parser.addoption("--generate-reference", action="store_true", default=False,
+                     help="execute integration test run to generate reference data")
+
+
 def pytest_configure(config):
     integration_tests_configpath = config.getvalue("integration-tests")
-    if integration_tests_configpath is not None:
+    if integration_tests_configpath:
         integration_tests_configpath = os.path.expandvars(
             os.path.expanduser(integration_tests_configpath)
         )
         config.option.integration_tests_config = yaml.load(
             open(integration_tests_configpath))
 
-        # Used by DokuReport class to show build environment details in report.
-        config._environment = []
-        # prevent opening dokupath on slave nodes (xdist)
-        if not hasattr(config, 'slaveinput'):
-            config.dokureport = DokuReport(
-                config.option.integration_tests_config['dokuwiki'])
-            config.pluginmanager.register(config.dokureport)
+        if not config.getoption("--generate-reference"):
+            # Used by DokuReport class to show build environment details in report.
+            config._environment = []
+            # prevent opening dokupath on slave nodes (xdist)
+            if not hasattr(config, 'slaveinput'):
+                config.dokureport = DokuReport(
+                    config.option.integration_tests_config['dokuwiki'])
+                config.pluginmanager.register(config.dokureport)
 
 
 def pytest_unconfigure(config):
+    # Unregister only if it was registered in pytest_configure
     integration_tests_configpath = config.getvalue("integration-tests")
-    if integration_tests_configpath is not None:
+    if integration_tests_configpath and not config.getoption("--generate-reference"):
         config.pluginmanager.unregister(config.dokureport)
 
 
+def pytest_terminal_summary(terminalreporter):
+    if terminalreporter.config.getoption("--generate-reference") and (
```

Comment:
Oh okay, I think I should have a close look at these very micro lints :grinning: